### PR TITLE
Implement Linkedlist class and notification function.

### PR DESCRIPTION
**Pull request by lisasasasa**
In this PR, I finished :
1. Add a directory `utility` in common
2. Implement Linkedlist data structure 
3. Implement the `showNotification` function in main UI with Linkedlist class.
4. Draw DOMElement on map

(TODO) notification's css

### CODE REVIEW

**Comment by lisasasasa** [comment]
不確定形象組的 UI 設計。
1. 我先讓 notification 是會在 div 中一直長出來的新 dom 
2. announcement 是跑馬燈，我還在想怎麼辦 ( 但我不太確定用 css 控制 marquee，在 multiple text 時，是不是好方法 )

**Review by fanlan1210 — COMMENTED** [review]
it seems OK, but marquee element is deprecated, may alter it by using div with CSS and JS.

File: sites/game-client/ui/main-ui.mjs [hunk 1]
```diff
@@ -58,14 +71,72 @@ class MainUI {
...
+      }
+    };
+
+    if (this._announceList.length === NOTIFICATION_NUM_LIMIT) {
+      removeNotification(this._announceList.findNode(NOTIFICATION_NUM_LIMIT), true);
+    }
+
+    const content = {msg};
+    const listNode = this._announceList.insert(content);
+    this.announcementWord.textContent = this._announceList.findAllElement().join('  ');
```
Comment (john0312) [inline sites/game-client/ui/main-ui.mjs:135]: Nit, optional: Separate this findAllElement().join() part into a separate function so we'll repeat lesser code?

File: sites/game-client/client.ejs [hunk 2]
```diff
@@ -35,13 +35,16 @@ absolute URL.
             <p id="error-message"></p>
           </div>
           <p id="error-handler-message"></p>
-        </div> 
+        </div>
         <% inDiv.forEach((i) => { %>
           <%- include(i); %>
         <% }); %>
       </div>
       <div id="notification-bar">
```
Comment (john0312) [inline sites/game-client/client.ejs:43]: Not really sure if we'll have a notification "bar" with flyout notification. We'll revisit this when the design is out.

**Review by john0312 — APPROVED** [review]
Given the amount of notifications that we might have (<10), I think we can consider using just plain array as well.

**Pull request merged**